### PR TITLE
Docs: Update SAML docs after the integration with SSO settings API

### PR DESCRIPTION
**Pull request by dmihai**
**What is this feature?**

Update the SAML docs after adding support for SAML in SSO settings API. The docs now mention the other methods available for configuring SAML: terraform and API.

**Why do we need this feature?**

For documenting new features available for SAML.

**Who is this feature for?**

Enterprise users using SAML.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md [hunk 1]
```diff
@@ -27,11 +27,21 @@ Available in [Grafana Enterprise]({{< relref "../../../../introduction/grafana-e
...
-Both methods offer the same configuration options, but you might prefer using the Grafana configuration file if you want to keep all of Grafana's authentication settings in one place. Grafana Cloud users do not have access to Grafana configuration file, so they should configure SAML through Grafana's UI.
+You can configure SAML authentication in Grafana through one of the following methods:
+
+- the Grafana configuration file
+- the API (refer to [SSO Settings API]({{< relref "../../../../developers/http_api/sso-settings" >}}))
+- the user interface (refer to [Configure SAML authentication using the Grafana user interface]({{< relref "../saml-ui" >}}))
+- the terraform provider (refer to [Terraform docs](https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/sso_settings))
+
+{{% admonition type="note" %}}
+The API and terraform are available in Public Preview in Grafana v11.1 behind the `ssoSettingsSAML` feature toggle.
```
Comment (mgyongyosi) [inline docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md:38]: Could you please also add that `ssoSettingsApi` should also be enabled? (iirc that is required)

File: docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md [hunk 2]
```diff
@@ -27,11 +27,21 @@ Available in [Grafana Enterprise]({{< relref "../../../../introduction/grafana-e
...
 
-You can configure SAML authentication in Grafana through the user interface (UI) or the Grafana configuration file. For instructions on how to set up SAML through Grafana's UI, refer to [Configure SAML authentication using the Grafana user interface]({{< relref "../saml-ui" >}}).
-Both methods offer the same configuration options, but you might prefer using the Grafana configuration file if you want to keep all of Grafana's authentication settings in one place. Grafana Cloud users do not have access to Grafana configuration file, so they should configure SAML through Grafana's UI.
+You can configure SAML authentication in Grafana through one of the following methods:
+
+- the Grafana configuration file
+- the API (refer to [SSO Settings API]({{< relref "../../../../developers/http_api/sso-settings" >}}))
+- the user interface (refer to [Configure SAML authentication using the Grafana user interface]({{< relref "../saml-ui" >}}))
+- the terraform provider (refer to [Terraform docs](https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/sso_settings))
```
Comment (mgyongyosi) [inline docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md:35]:
```suggestion
- the Terraform provider (refer to [Terraform docs](https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/sso_settings))
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md (hunk 1 above)
Comment (mgyongyosi) [inline docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md:38]:
```suggestion
The API and Terraform support are available in Public Preview in Grafana v11.1 behind the `ssoSettingsSAML` feature toggle.
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md [hunk 3]
```diff
@@ -27,11 +27,21 @@ Available in [Grafana Enterprise]({{< relref "../../../../introduction/grafana-e
...
+
+{{% admonition type="note" %}}
+The API and terraform are available in Public Preview in Grafana v11.1 behind the `ssoSettingsSAML` feature toggle.
+{{% /admonition %}}
+
+All methods offer the same configuration options, but you might prefer using the Grafana configuration file or the terraform provider if you want to keep all of Grafana's authentication settings in one place. Grafana Cloud users do not have access to Grafana configuration file, so they should configure SAML through the other methods.
```
Comment (mgyongyosi) [inline docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md:41]:
```suggestion
All methods offer the same configuration options, but you might prefer using the Grafana configuration file or the Terraform provider if you want to keep all of Grafana's authentication settings in one place. Grafana Cloud users do not have access to Grafana configuration file, so they should configure SAML through the other methods.
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md (hunk 1 above)
Comment (dmihai) [reply docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md]: Done.

File: docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md (hunk 2 above)
Comment (dmihai) [reply docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md]: Done.

File: docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md (hunk 1 above)
Comment (dmihai) [reply docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md]: Done.

File: docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md (hunk 3 above)
Comment (dmihai) [reply docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md]: Done.

**Comment by mgyongyosi** [comment]
@chri2547 Could you please review this? Thanks!

File: docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md [hunk 4]
```diff
@@ -27,11 +27,21 @@ Available in [Grafana Enterprise]({{< relref "../../../../introduction/grafana-e
...
-Both methods offer the same configuration options, but you might prefer using the Grafana configuration file if you want to keep all of Grafana's authentication settings in one place. Grafana Cloud users do not have access to Grafana configuration file, so they should configure SAML through Grafana's UI.
+You can configure SAML authentication in Grafana through one of the following methods:
+
+- the Grafana configuration file
+- the API (refer to [SSO Settings API]({{< relref "../../../../developers/http_api/sso-settings" >}}))
+- the user interface (refer to [Configure SAML authentication using the Grafana user interface]({{< relref "../saml-ui" >}}))
+- the Terraform provider (refer to [Terraform docs](https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/sso_settings))
+
+{{% admonition type="note" %}}
+The API and Terraform support are available in Public Preview in Grafana v11.1 behind the `ssoSettingsSAML` feature toggle. The `ssoSettingsApi` flag also has to be enabled.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md:38]:
```suggestion
The API and Terraform support are available in Public Preview in Grafana v11.1 behind the `ssoSettingsSAML` feature toggle. You must also enable the `ssoSettingsApi` flag.
```

Comment (dmihai) [reply docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md]: Done, thanks for reviewing!

**Review by chri2547 — APPROVED** [review]
Docs looks good! Thank you for the contribution!

**Pull request merged**